pkg/cluster/operation/destroy.go

```diff
@@ -187,7 +290,7 @@ func DestroyComponent(getter ExecutorGetter, instances []spec.Instance, cls spec
 	for _, ins := range instances {
 		// Some data of instances will be retained
 		dataRetained := retainDataRoles.Exist(ins.ComponentName()) ||
-			retainDataNodes.Exist(ins.ID())
+			retainDataNodes.Exist(ins.ID()) || retainDataNodes.Exist(ins.GetHost())
```

Comment:
If we apply this change, we should change the documentation, this feature allows user to retain data by the host.